cupyx/scipy/sparse/data.py

```diff
@@ -88,6 +89,219 @@ def power(self, n, dtype=None):
+            mat.sum_duplicates()
+
+        # Do the reudction
+        value = mat._minor_reduce(min_or_max, axis, nonzero)
+
+        return value
+
+    def _min_or_max(self, axis, out, min_or_max, sum_duplicates, non_zero):
+        if out is not None:
+            raise ValueError(("Sparse matrices do not support "
+                              "an 'out' parameter."))
+
+        validateaxis(axis)
+
+        if axis == 0 or axis == 1:
+            return self._min_or_max_axis(axis, min_or_max, sum_duplicates,
+                                         non_zero)
+        else:
+            raise ValueError("axis out of range")
+
+    def _arg_min_or_max_axis(self, axis, op, sum_duplicates):
+        if self.shape[axis] == 0:
+            raise ValueError("Can't apply the operation along a zero-sized "
+                             "dimension.")
+
+        mat = self.tocsc() if axis == 0 else self.tocsr()
+
+        if sum_duplicates:
+            mat.sum_duplicates()
+
+        # Do the reudction
+        value = mat._arg_minor_reduce(op, axis)
+
+        return value
+
+    def _arg_min_or_max(self, axis, out, op, compare, sum_duplicates):
+        if out is not None:
+            raise ValueError("Sparse matrices do not support "
+                             "an 'out' parameter.")
+
+        validateaxis(axis)
+
+        if axis is None:
+            if 0 in self.shape:
+                raise ValueError("Can't apply the operation to "
+                                 "an empty matrix.")
+
+            if self.nnz == 0:
+                return 0
+            else:
+                zero = self.dtype.type(0)
+                mat = self.tocoo()
+
+                if sum_duplicates:
+                    mat.sum_duplicates()
+
+                am = op(mat.data)
+                m = mat.data[am]
+
+                if compare(m, zero):
+                    return mat.row[am] * mat.shape[1] + mat.col[am]
+                else:
+                    size = cupy.prod(mat.shape)
+                    if size == mat.nnz:
+                        return am
+                    else:
+                        ind = mat.row * mat.shape[1] + mat.col
+                        zero_ind = _find_missing_index(ind, size)
+                        if m == zero:
+                            return min(zero_ind, am)
+                        else:
+                            return zero_ind
+
+        return self._arg_min_or_max_axis(axis, op, sum_duplicates)
+
+    def max(self, axis=None, out=None, sum_duplicates=False, nonzero=False):
+        """
+        Returns the maximum of the matrix or maximum along an axis.
+        This takes all elements into account, not just the non-zero ones.
+
+        Args:
+            axis: {-2, -1, 0, 1, None} (optional)
+                Axis along which the sum is computed. The default is to
+                compute the maximum over all the matrix elements, returning
+                a scalar (i.e. `axis` = `None`).
```

Comment:
cupy format is \`\`axis\`\` or  \`\`None\`\` instead of \`axis\` and \`None\`

Reply:
This should be corrected.